tests/components/tradfri/test_cover.py

```diff
@@ -0,0 +1,219 @@
+"""Tradfri cover (recognised as blinds in the IKEA echosystem) platform tests."""
+
+from copy import deepcopy
+from unittest.mock import MagicMock, Mock, PropertyMock, patch
+
+import pytest
+from pytradfri.device import Device
+from pytradfri.device.blind import Blind
+from pytradfri.device.blind_control import BlindControl
+
+from homeassistant.components import tradfri
+
+from . import GATEWAY_ID
+
+from tests.common import MockConfigEntry
+
+SET_POSITION_TEST_CASES = [
+    # Open cover
+    [{}, {"position": 100}, {"state": "open"}],
+    # Close cover
+    [{}, {"position": 0}, {"state": "closed"}],
+]
+
+
+@pytest.fixture(autouse=True, scope="module")
+def setup(request):
+    """Set up patches for pytradfri methods."""
+    p_1 = patch(
+        "pytradfri.device.BlindControl.raw",
+        new_callable=PropertyMock,
+        return_value=[{"mock": "mock"}],
+    )
+    p_2 = patch("pytradfri.device.BlindControl.blinds")
+    p_1.start()
+    p_2.start()
+
+    def teardown():
+        """Remove patches for pytradfri methods."""
+        p_1.stop()
+        p_2.stop()
+
+    request.addfinalizer(teardown)
+
+
+async def generate_psk(self, code):
+    """Mock psk."""
+    return "mock"
+
+
+async def setup_integration(hass):
```

Comment:
We shouldn't duplicate code that is available already in the light tests. Move common tools to a common module in the test package.